### PR TITLE
MAM-4034-cant-interact-with-remote-posts-in-akkoma

### PR DESCRIPTION
**Pull request by derspyy**

### CODE REVIEW

File: Mammoth/Services/Backend/StatusService.swift [hunk 1]
```diff
@@ -495,10 +495,9 @@ extension StatusService {
         } catch let error {
             do {
                 switch error as? ClientError {
-                case .mastodonError(let message):
-                    if message == "Record not found" {
-                        return try await self.runTaskLocally(forStatus: status, task: task)
-                    }
+                case .mastodonError:
+                    // these error messages can be localized.
```
Comment (bnolens) [inline Mammoth/Services/Backend/StatusService.swift:499]: Are there other mastodon errors than the `Record not found` one? I would guess there are. 
In this scenario, we only want to run the task locally if the post is not found. You must keep this check - by looking at an error code instead? Or is there an error key in this error object we could use?

**Review by bnolens — COMMENTED** [review]
Please have a look at my comment

File: Mammoth/Services/Backend/StatusService.swift (hunk 1 above)
Comment (derspyy) [reply Mammoth/Services/Backend/StatusService.swift:499]: we could look at the 404 itself. wdyt?

Comment (bnolens) [reply Mammoth/Services/Backend/StatusService.swift:499]: Might work yes. Is a `Record not found` always paired with a 404? Can 404 happen in other scenarios?
@rileyhCode do you think checking the 404 error code is enough here?

Comment (rileyhCode) [reply Mammoth/Services/Backend/StatusService.swift:499]: Hm. It looks like "Record not found" is coming from the data returned (MastodonError type, json "error" field).
Anyway… checking for 404 would probably work (but should be tested, obv…)

Comment (derspyy) [reply Mammoth/Services/Backend/StatusService.swift:499]: not an easy way to do this using the current library...

**Comment by derspyy** [comment]
not ready for merge BTW

**Pull request closed without merging**